test/e2e/storage/persistent_volumes-local.go

```diff
@@ -252,6 +253,96 @@ var _ = SIGDescribe("PersistentVolumes-local [Feature:LocalPersistentVolumes] [S
 		})
 	})
 
+	Context("when pod using local volume with non-existant path", func() {
+		It("should not be able to mount", func() {
+			for _, testVolType := range LocalVolumeTypes {
+				By(fmt.Sprintf("local-volume-type: %s", testVolType))
+				testVol := &localTestVolume{
+					node:            config.node0,
+					hostDir:         "/non-existent/location/nowhere",
+					localVolumeType: testVolType,
+				}
+				By("Creating local PVC and PV")
+				createLocalPVCPV(config, testVol)
+				pod, _ := createLocalPod(config, testVol)
+				var events *v1.EventList
+				selector := fields.Set{
+					"involvedObject.kind":      "Pod",
+					"involvedObject.name":      pod.Name,
+					"involvedObject.namespace": config.ns,
+					"reason":                   "FailedMount",
+				}.AsSelector().String()
+				options := metav1.ListOptions{FieldSelector: selector}
+				events, err := config.client.Core().Events(config.ns).List(options)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(len(events.Items)).NotTo(Equal(0))
+				Expect(events.Items[0].Message).To(ContainSubstring("MountVolume.SetUp failed"))
+				Expect(events.Items[0].Message).To(ContainSubstring("does not exist"))
+			}
+		})
+	})
+	Context("when pod has nodeName set to a different node than the PV node affinity", func() {
+		It("should not be able to mount", func() {
```

Comment:
Can you make the 3 ways to specify a node in a Pod each an It() under this Context()?